plugins/scram.c

```diff
@@ -1281,6 +1301,14 @@ scram_server_mech_step2(server_context_t *text,
 	    goto cleanup;
 	}
 
+	if (sparams->cbinding == NULL ||
+	    strcmp(sparams->cbinding->name, text->cbindingname) != 0) {
+	    /* Try to get desired channel binding data */
+	    scram_server_get_channel_binding(sparams,
+					     scram_sasl_mech,
+					     text->cbindingname);
```

Comment:
Similarly here: you should test the result of the new function call and return error if it is not SASL_OK.

Reply:
Same here. Error handling is done in the follow-up code. Shall I change the function declaration in static ***void*** scram_server_get_channel_binding(...)

Reply:
scram_server_get_channel_binding now ignores the callback result. Error handling is done by protocol if callback cannot set the correct channel binding data.